<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="MacBook-Air.local" failures="1" tests="14" name="testcases.LoginTest" time="91.117" errors="1" timestamp="2025-05-06T18:15:09 IST" skipped="0">
  <testcase classname="testcases.LoginTest" name="TC01_VerifyAllFieldsDisplayed" time="4.458"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC02_VerifyPlaceholderText" time="4.248"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC03_LoginWithValidCredentials" time="10.491"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC04_LoginWithInvalidCredentials" time="6.586">
    <error message="Cannot find elements when the XPath is null." type="java.lang.IllegalArgumentException">
      <![CDATA[java.lang.IllegalArgumentException: Cannot find elements when the XPath is null.
at org.openqa.selenium.internal.Require$ArgumentChecker.nonNull(Require.java:214)
at org.openqa.selenium.By$ByXPath.<init>(By.java:286)
at org.openqa.selenium.By.xpath(By.java:89)
at Pages.BasePage.getElement(BasePage.java:26)
at testcases.LoginTest.TC04_LoginWithInvalidCredentials(LoginTest.java:86)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </error>
  </testcase> <!-- TC04_LoginWithInvalidCredentials -->
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC05_LoginWithEmptyFields" time="4.848"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC06_LoginWithEmptyUsername" time="4.686"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC07_LoginWithEmptyPassword" time="4.576"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC08_VerifyPasswordMasked" time="4.440"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC09_RememberMeCheckboxFunctionality" time="10.174">
    <failure message="Username is not remembered expected [abhilasha01] but found []" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Username is not remembered expected [abhilasha01] but found []
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertEqualsImpl(Assert.java:140)
at org.testng.Assert.assertEquals(Assert.java:122)
at org.testng.Assert.assertEquals(Assert.java:629)
at testcases.LoginTest.TC09_RememberMeCheckboxFunctionality(LoginTest.java:132)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
    </failure>
  </testcase> <!-- TC09_RememberMeCheckboxFunctionality -->
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC10_LoginWithoutSelectingRememberMe" time="11.233"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC13_VerifyErrorMessageDisappearsAfterCorrectingInput" time="9.675"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC14_VerifyCheckboxDefaultState" time="4.247"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC17_VerifyLoginWithSpacesBeforeAfterUsername" time="6.941"/>
  <system-out/>
  <testcase classname="testcases.LoginTest" name="TC18_VerifyLoginWithSpacesBeforeAfterPassword" time="4.514"/>
  <system-out/>
</testsuite> <!-- testcases.LoginTest -->
